Задача 1 A. Закраска прямой

На числовой прямой окрасили N отрезков. Известны координаты левого и правого концов каждого отрезка (Li и Ri). Найти длину окрашенной части числовой прямой.

Формат ввода
В первой строке находится число N, в следующих N строках - пары Li и Ri. Li и Ri - целые, -109 ≤ Li ≤ Ri ≤ 109, 1 ≤ N ≤ 15 000

Формат вывода
Вывести одно число - длину окрашенной части прямой.

In [ ]:
1
10 20

10

1 
10 10

0

2
10 20
20 40

30

In [247]:
n = int(input())


3


In [297]:
events = [0]*n
for i in range(n):
    l, r = map(int, input().split()) 
    events[i] = [l ,r]
events.sort()

0 0 
10 20
15 20


In [293]:
events

[[0, 0], [10, 20], [15, 20]]

In [295]:
long = 0
for i in range(len(events)):
    print(events[i])
    if i > 0 and events[i][0] < events[i - 1][1] and events[i][1] > events[i - 1][1]:
        print('if')
        events[i][0] = events[i - 1][1]
        
        
    elif i > 0 and events[i][0] < events[i - 1][1] and events[i][1] <= events[i - 1][1]:
        print('elif')
        events[i][0] = events[i - 1][0]
        if events[i][1] <= events[i - 1][1]:
            events[i][1] = events[i - 1][1]
        
        
    if i > 0 and events[i][1] == events[i - 1][1] and events[i][0] == events[i - 1][0]:
        i += 1
    else:
        long +=  events[i][1] - events[i][0]
        
    
        
    
    print('long',long)

[0, 0]
long 0
[10, 20]
long 10
[15, 20]
elif
long 10


In [ ]:
n = int(input())
events = [0]*n
for i in range(n):
    l, r = map(int, input().split()) 
    events[i] = [l ,r]

events.sort()

long = 0
for i in range(len(events)):
    
    if i > 0 and events[i][0] < events[i - 1][1] and events[i][1] > events[i - 1][1]:
        
        events[i][0] = events[i - 1][1]
        
        
    elif i > 0 and events[i][0] < events[i - 1][1] and events[i][1] <= events[i - 1][1]:
        
        events[i][0] = events[i - 1][0]
        if events[i][1] <= events[i - 1][1]:
            events[i][1] = events[i - 1][1]
        
        
    if i > 0 and events[i][1] == events[i - 1][1] and events[i][0] == events[i - 1][0]:
        i += 1
    else:
        long +=  events[i][1] - events[i][0]
        
    
        
    
print(long)

Задача 2 B. Таможня

 список из N грузов, которые будут контролироваться Министерством. Для каждого груза известны время его прибытия, отсчитываемое с некоторого момента, хранимого в большом секрете, и время, требуемое аппарату для обработки этого груза. Шеф дал Мише задание по этим данным определить, какое минимальное количество аппаратов необходимо заказать на заводе, чтобы все грузы Министерства начинали досматриваться сразу после прибытия. Необходимо учесть, что конструкция тех аппаратов, которые было решено установить, не позволяет обрабатывать два груза одновременно на одном аппарате.
    Формат ввода
На первой строке входного файла задано число N (0 ≤ N ≤ 50 000). На следующих N строках находится по 2 целых положительных числа Ti и Li – время прибытия соответствующего груза и время, требуемое для его обработки (1 ≤ Ti ≤ 106, 1 ≤ Li ≤ 106).

Формат вывода
В выходной файл выведите одно число – наименьшее количество аппаратов, которое нужно установить, чтобы не вызвать подозрений у Министерства.

In [ ]:
3
3 2
4 2
5 2

2

5
10 3
11 5
12 3
15 1
13 4

3

10
79 55
81 84
34 11
59 57
98 96
98 59
27 59
55 1
36 38
26 60

In [169]:
a = n
c = {}
for i in range(n):
    print('============')
    print('i: ',i)
    
    if i > 0:
        
        j = -1
        while j < i:
            j += 1
            print('j: ',j)
            print(events[j][1])
            print(events[i][0])
            print(events[j][1],',', events[i][0])
            
            if events[j][2] <= events[i][0]:
                
                print('TRUE --------', events[j][2], 'j',j)
                a -= 1
                break
            
print('число аппаратов',a)        

i:  0
i:  1
j:  0
8
166
8 , 166
TRUE -------- 97 j 0
i:  2
j:  0
8
262
8 , 262
TRUE -------- 97 j 0
i:  3
j:  0
8
294
8 , 294
TRUE -------- 97 j 0
i:  4
j:  0
8
473
8 , 473
TRUE -------- 97 j 0
i:  5
j:  0
8
650
8 , 650
TRUE -------- 97 j 0
i:  6
j:  0
8
762
8 , 762
TRUE -------- 97 j 0
i:  7
j:  0
8
814
8 , 814
TRUE -------- 97 j 0
i:  8
j:  0
8
891
8 , 891
TRUE -------- 97 j 0
i:  9
j:  0
8
950
8 , 950
TRUE -------- 97 j 0
число аппаратов 1


In [252]:
n = int(input())

10


In [253]:
def max_visit_online(n, tin, tout):
    events = [0] * (2 * n )

    print(len(events))
    for i in range(n ):
        print(i)
        print(i % 2)
        t, l = map(int, input().split()) 
        
        events[2*i] = (t, 1)
        print('+',2*i,events[i])
        
        events[2*i+1] = (t+l, -1)
        print('-',2*i+1,events[i])
        print(events)
    events.sort()
    print(events)
    a = 0
    maxonline = 0
    for event in events:
        if event[1] == -1:
            a -= 1
        else:
            a += 1
        print('o',a)
        maxonline = max(a, maxonline)
        
    return maxonline

In [255]:
max_visit_online(n)

20
0
0
79 55
+ 0 (79, 1)
- 1 (79, 1)
[(79, 1), (134, -1), 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
1
1
 81 84
+ 2 (134, -1)
- 3 (134, -1)
[(79, 1), (134, -1), (81, 1), (165, -1), 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
2
0
 34 11
+ 4 (81, 1)
- 5 (81, 1)
[(79, 1), (134, -1), (81, 1), (165, -1), (34, 1), (45, -1), 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
3
1
 59 57
+ 6 (165, -1)
- 7 (165, -1)
[(79, 1), (134, -1), (81, 1), (165, -1), (34, 1), (45, -1), (59, 1), (116, -1), 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
4
0
 98 96
+ 8 (34, 1)
- 9 (34, 1)
[(79, 1), (134, -1), (81, 1), (165, -1), (34, 1), (45, -1), (59, 1), (116, -1), (98, 1), (194, -1), 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
5
1
 98 59
+ 10 (45, -1)
- 11 (45, -1)
[(79, 1), (134, -1), (81, 1), (165, -1), (34, 1), (45, -1), (59, 1), (116, -1), (98, 1), (194, -1), (98, 1), (157, -1), 0, 0, 0, 0, 0, 0, 0, 0]
6
0
 27 59
+ 12 (59, 1)
- 13 (59, 1)
[(79, 1), (134, -1), (81, 1), (165, -1), (34, 1), (45, -1), (59, 1), (116, -1), (98

5

762 4
294 1
814 3
891 9
166 8
89 8
650 10
262 6
473 2
950 3

In [256]:

n = int(input())
events = [0] * (2 * n)


for i in range(n):

    t, l = map(int, input().split()) 
        
    events[2*i] = (t, 1)
    events[2*i+1] = (t+l, -1)

events.sort()
    
a = 0
maxonline = 0
for event in events:
    if event[1] == -1:
        a -= 1
    else:
        a += 1
    
    maxonline = max(a, maxonline)
print(maxonline)
        


3
3 2
4 2
5 2
2


Задача 3 C. Минимальное покрытие

На прямой задано некоторое множество отрезков с целочисленными координатами концов [Li, Ri]. Выберите среди данного множества подмножество отрезков, целиком покрывающее отрезок [0, M], (M — натуральное число), содержащее наименьшее число отрезков.

Формат ввода
В первой строке указана константа M (1 ≤ M ≤ 5000). В каждой последующей строке записана пара чисел Li и Ri (Li, Ri ≤ 50000), задающая координаты левого и правого концов отрезков. Список завершается парой нулей. Общее число отрезков не превышает 100 000.

Формат вывода
В первой строке выходного файла выведите минимальное число отрезков, необходимое для покрытия отрезка [0; M]. Далее выведите список покрывающего подмножества, упорядоченный по возрастанию координат левых концов отрезков. Список отрезков выводится в том же формате, что и во входe. Завершающие два нуля выводить не нужно. Если покрытие отрезка [0, M] исходным множеством отрезков [Li, Ri] невозможно, то следует вывести единственную фразу “No solution”.

In [339]:
M = int(input())
O = 0

500


In [342]:
events = []
i = 0
while True:

    l, r = map(int, input().split()) 
    if l == 0 and r == 0 :
        break
    events.append((l, r))
    
events.sort()

-1000 13
 5000 50000
 15 499
 13 18
 18 500
 15 18
 -5000 5
 -8000 -3
0 0


In [343]:
events = a

In [346]:
c = []
max_r_idx = 0
l = r = 0
i = 0
for i in range(len(events)):
    print(events[i][0] ,l,events[i][1],r )
    if events[i][0] <= l and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:
        print('--------------')
        max_r_idx = i
    

if events[max_r_idx][0] <= l and events[max_r_idx][1] >= r:
    c.append(max_r_idx)
print('c',c)
l = events[max_r_idx][0]
r = events[max_r_idx][1]
print('out l,r',l,r)


-8000 0 -3 0
-5000 0 5 0
--------------
-1000 0 13 0
--------------
13 0 18 0
15 0 18 0
15 0 499 0
18 0 500 0
5000 0 50000 0
c [2]
out l,r -1000 13


In [336]:
while i < len(events) and r < M:

    for i in range(1,len(events)):
        
        #print(events[i][0] ,events[i][1] )
    
        if events[i][0] >= l and events[i][0] <= r and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:
            print('--------------')
            max_r_idx = i
            print(max_r_idx)
            print('--------------')
    print(c[-1],max_r_idx)
    if max_r_idx == c[-1]:
        break
    print('max_r_idx',max_r_idx)
    c.append(max_r_idx)
    print('c',c)
    l = events[max_r_idx][0]
    r = events[max_r_idx][1]
    print('out l,r',l,r)
if r >= M and c != []:
    print(len(c))
    for i in c:
        print(*events[i])
else:
    print('No solution')

No solution


In [214]:
c = []
for i in range(len(events)):
    print(events[i][0],events[i][1])
    if events[i][0] <= l and events[i][1] >= r:
        c.append(i)
print(c)
if c == []:
    print('No solution')

-3 -2
-2 2
-1 3
1 5
2 7
4 6
5 8
6 10
7 10
[2]


In [208]:
c = a

In [ ]:
max_r_idx = c[0]

In [218]:

print(max_r_idx)
for i in c:
    if events[i][1] > events[max_r_idx][1] and events[i][1] < M:
        max_r_idx = i
    if events[i][1] >= M:
        break
        
l = events[max_r_idx][0]
r = events[max_r_idx][1]
print(l,r)

2
-1 3


In [217]:
c = []
for i in range(len(events)):
    print(events[i][0],events[i][1])
    if events[i][0] <= l and events[i][1] >= r:
        c.append(i)
print(c)


-3 -2
-2 2
-1 3
1 5
2 7
4 6
5 8
6 10
7 10
[2]


In [265]:
c = []
max_r_idx = 0
l = r = 0
i = 0

while r < M:
    for i in range(len(events)):
        print('in l,r',l,r)
        print(events[i][0] ,events[i][1] )
    
        if events[i][0] <= l and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:
            print('--------------')
            max_r_idx = i
    

    print('max_r_idx',max_r_idx)
    c.append(max_r_idx)
    print('c',c)
    l = events[max_r_idx][0]
    r = events[max_r_idx][1]
    print('out l,r',l,r)

in l,r 0 0
-3 -2
in l,r 0 0
-2 2
--------------
in l,r 0 0
-1 3
--------------
in l,r 0 0
1 5
in l,r 0 0
2 7
in l,r 0 0
4 6
in l,r 0 0
5 8
in l,r 0 0
6 10
in l,r 0 0
7 10
max_r_idx 2
c [2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1

in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
i

in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_i

in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-

5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,

in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 

in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,

in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
out l,r -1 3
in l,r -1 3
-3 -2
in l,r -1 3


in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

-3 -2
in l,r -1 3
-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 

4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,

1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,

-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

-2 2
in l,r -1 3
-1 3
in l,r -1 3
1 5
in l,r -1 3
2 7
in l,r -1 3
4 6
in l,r -1 3
5 8
in l,r -1 3
6 10
in l,r -1 3
7 10
max_r_idx 2
c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

c [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2

KeyboardInterrupt: 

In [351]:
M = int(input())
events = []
i = 0
while True:

    l, r = map(int, input().split()) 
    if l == 0 and r == 0 :
        break
    events.append((l, r))
    
events.sort()

c = []
max_r_idx = 0
l = r = 0

for i in range(len(events)):
    
    if events[i][0] <= l and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:
        
        max_r_idx = i
    
if events[max_r_idx][0] <= l and events[max_r_idx][1] >= r:
    c.append(max_r_idx)

    l = events[max_r_idx][0]
    r = events[max_r_idx][1]

while i < len(events) and r < M:
    for i in range(1,len(events)):
        
        if events[i][0] >= l and events[i][0] <= r and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:           
            max_r_idx = i
            
    
    if max_r_idx == c[-1]:
        break
    
    c.append(max_r_idx)
    
    l = events[max_r_idx][0]
    r = events[max_r_idx][1]
    
if r >= M and c != []:
    print(len(c))
    for i in c:
        print(*events[i])
else:
    print('No solution')

1
-1 0
0 1
0 0
1
0 1


In [ ]:
M = int(input())
events = []
i = 0
while True:

    l, r = map(int, input().split()) 
    if l == 0 and r == 0 :
        break
    events.append((l, r))
    
events.sort()

c = []
max_r_idx = 0
l = r = 0

for i in range(len(events)):
    
    if events[i][0] <= l and events[i][1] >= r and events[i][1] > events[max_r_idx][1]:
        
        max_r_idx = i
if   events[max_r_idx][0] <= 0:  
	c.append(max_r_idx)

	l = events[max_r_idx][0]
	r = events[max_r_idx][1]

while i < len(events) and r < M:
    if c == []:
        break
    for i in range(max_r_idx,len(events)):
        
        if events[i][0] >= l and events[i][0] <= r and events[i][1] >= r and events[i][1] > events[max_r_idx][1] :
            
            max_r_idx = i
            
    
    if max_r_idx == c[-1]:
        break
    
    c.append(max_r_idx)
    
    l = events[max_r_idx][0]
    r = events[max_r_idx][1]
    
if r >= M and c != []:
    print(len(c))
    for i in c:
        print(*events[i])
else:
	print('No solution')

# 500
-1000 13
5000 50000
15 499
13 18
18 500
15 18
-5000 5
-8000 -3
0 0

Задача 4 D. Наполненность котятами

На прямой в точках 
a1,a2,…,an (возможно, совпадающих) сидят n котят. На той же прямой лежат m отрезков [l1,r1],[l2,r2],…,[lm,rm]. Нужно для каждого отрезка узнать его наполненность котятами — сколько котят сидит на отрезке.

Формат ввода
На первой строке n и m (1≤n,m≤105). На второй строке n целых чисел ai (0≤ai≤109). Следующие m строк содержат пары целых чисел li,ri (0≤li≤ri≤109).

Формат вывода
Выведите m целых чисел. 
i-е число — наполненность котятами 
i-го отрезка.

Задача 5 E. Полярные прямоугольники

Формат ввода
В первой строке вводится целое число N — количество прямоугольников (1 ≤ N ≤ 100 000). Далее в N строках содержится описание прямоугольников. Каждый прямоугольник описывается четверкой действительных чисел r1, r2, φ1, φ2, где r1, r2 обозначают радиусы окружностей, образующих кольцо (r1 < r2), а φ1, φ2 обозначают углы, образованные первым и вторым лучами с осью абсцисс, заданные в радианах. При этом заметается область от первого луча до второго в направлении против часовой стрелки (т.е. возрастания углов), даже в случае, когда φ1 > φ2. Все числа заданы максимум с шестью знаками после десятичной точки. Углы лежат в полуинтервале [0, 2π), а радиусы не превосходят 106. Гарантируется, что φ1 ≠ φ2.

Формат вывода
Выведите единственное число — площадь искомого пересечения. Ответ будет считаться правильным, если его абсолютная или относительная погрешность не будет превышать 10-6.

2

1 3 0 3

2 4 1.5 4.5

ВЫВОД
3.7500000000

In [1]:
N = int(input())

2


In [30]:
S = 0
events = []
for i in range(1, N + 1):
    r, R, fi_1, fi_2 = map(float, input().split()) 
    events.append((fi_1, -i))
    events.append((fi_2, i))
    if i == 1:
        max_r = r
        min_R = R
    else:
        max_r = max(max_r, r)
        min_R = min(min_R, R)
events.sort()
print(events, '\n',min_R,'\n',max_r)


1 2 0 3
1 2 2 1
[(0.0, -1), (1.0, 2), (2.0, -2), (3.0, 1)] 
 2.0 
 1.0


In [33]:
online = 0
used = set()
for event in events:
    print(event)
    
    if event[1] < 0:
        online += 1
        used.add(-event[-1])
    elif event[1] in used:
        online -= 1
    print('online',online)    
print(online,'\n',used)


(0.0, -1)
online 1
(1.0, 2)
online 1
(2.0, -2)
online 2
(3.0, 1)
online 1
1 
 {1, 2}


In [35]:
a = 0 

for i in range(len(events)):
    event = events[i]
    print('event',event)
    if event[1] < 0:
        online += 1
        print('+',online)
    else:
        online -= 1
        print('-',online)
    print('online',online)
    if online == N:
        if i < len(events) - 1:
            print('if')
            print(i+1,'fi2', i,'fi1')
            a += (events[i + 1][0] - events[i][0]) * ( min_R ** 2 - max_r ** 2) / 2
            
        else:
            print('else')
            print('fi2', len(events) - 1,'fi1')
            a += (events[0][0] - events[len(events) - 1][0] + 2 * pi) * ( min_R ** 2 - max_r ** 2) / 2
        print('a',a)
print('S: ',a)

event (0.0, -1)
+ 2
online 2
if
1 fi2 0 fi1
a 1.5
event (1.0, 2)
- 1
online 1
event (2.0, -2)
+ 2
online 2
if
3 fi2 2 fi1
a 3.0
event (3.0, 1)
- 1
online 1
S:  3.0


In [29]:
from math import pi
N = int(input())


events = []
for i in range(1, N + 1):
    r, R, fi_1, fi_2 = map(float, input().split()) 
    events.append((fi_1, -i))
    events.append((fi_2, i))
    if i == 1:
        max_r = r
        min_R = R
    else:
        max_r = max(max_r, r)
        min_R = min(min_R, R)
events.sort()

online = 0
used = set()

for event in events:
    
    
    if event[1] < 0:
        online += 1
        used.add(-event[1])
    elif event[1] in used:
        online -= 1

a = 0 

for i in range(len(events)):
    event = events[i]
    
    if event[1] < 0:
        online += 1
        
    else:
        online -= 1
        
    
    if online == N:
        if i < len(events) - 1:
            
            a += (events[i + 1][0] - events[i][0]) * ( min_R ** 2 - max_r ** 2) / 2
            
        else:
            
            a += (events[0][0] - events[len(events) - 1][0] + 2 * pi) * ( min_R ** 2 - max_r ** 2) / 2
        
print(a)


2
87.6 90.3 3.3 1.4
90.3 98.2 5.1 2.9
0.0
